### ISSUE TITLE
Crash on AppListActivity.onSaveInstanceState

### ISSUE THREAD

**Issue by yanokwa** [getodk/collect#1412]
```
java.lang.NullPointerException: 
  at org.odk.collect.android.activities.AppListActivity.onSaveInstanceState (AppListActivity.java:144)
  at org.odk.collect.android.activities.FormDownloadList.onSaveInstanceState (FormDownloadList.java:345)
  at android.app.Activity.performSaveInstanceState (Activity.java:1388)
  at android.app.Instrumentation.callActivityOnSaveInstanceState (Instrumentation.java:1286)
  at android.app.ActivityThread.callCallActivityOnSaveInstanceState (ActivityThread.java:4616)
  at android.app.ActivityThread.handleRelaunchActivity (ActivityThread.java:4570)
  at android.app.ActivityThread.access$1000 (ActivityThread.java:181)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1488)
  at android.os.Handler.dispatchMessage (Handler.java:102)
  at android.os.Looper.loop (Looper.java:145)
  at android.app.ActivityThread.main (ActivityThread.java:6145)
  at java.lang.reflect.Method.invoke (Native Method)
  at java.lang.reflect.Method.invoke (Method.java:372)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1399)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1194)
```

```
java.lang.NullPointerException: 
  at org.odk.collect.android.activities.AppListActivity.onSaveInstanceState (AppListActivity.java:144)
  at org.odk.collect.android.activities.FormChooserList.onSaveInstanceState (FormChooserList.java:103)
  at android.app.Activity.performSaveInstanceState (Activity.java:1307)
  at android.app.Instrumentation.callActivityOnSaveInstanceState (Instrumentation.java:1288)
  at android.app.ActivityThread.callCallActivityOnSaveInstanceState (ActivityThread.java:3958)
  at android.app.ActivityThread.performStopActivityInner (ActivityThread.java:3369)
  at android.app.ActivityThread.handleStopActivity (ActivityThread.java:3425)
  at android.app.ActivityThread.access$1100 (ActivityThread.java:151)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1327)
  at android.os.Handler.dispatchMessage (Handler.java:102)
  at android.os.Looper.loop (Looper.java:135)
  at android.app.ActivityThread.main (ActivityThread.java:5255)
  at java.lang.reflect.Method.invoke (Native Method)
  at java.lang.reflect.Method.invoke (Method.java:372)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:905)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:700)
```

```
java.lang.NullPointerException: 
  at org.odk.collect.android.activities.AppListActivity.onSaveInstanceState (AppListActivity.java:144)
  at org.odk.collect.android.activities.InstanceUploaderList.onSaveInstanceState (InstanceUploaderList.java:272)
  at android.app.Activity.performSaveInstanceState (Activity.java:1307)
  at android.app.Instrumentation.callActivityOnSaveInstanceState (Instrumentation.java:1288)
  at android.app.ActivityThread.callCallActivityOnSaveInstanceState (ActivityThread.java:3958)
  at android.app.ActivityThread.performStopActivityInner (ActivityThread.java:3369)
  at android.app.ActivityThread.handleStopActivity (ActivityThread.java:3425)
  at android.app.ActivityThread.access$1100 (ActivityThread.java:151)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1327)
  at android.os.Handler.dispatchMessage (Handler.java:102)
  at android.os.Looper.loop (Looper.java:135)
  at android.app.ActivityThread.main (ActivityThread.java:5255)
  at java.lang.reflect.Method.invoke (Native Method)
  at java.lang.reflect.Method.invoke (Method.java:372)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:905)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:700)
```

**Comment by yanokwa** [comment]
From firebase...

```
Exception java.lang.NullPointerException: Attempt to invoke virtual method 'boolean android.support.v7.widget.SearchView.isIconified()' on a null object reference
org.odk.collect.android.activities.AppListActivity.onSaveInstanceState (AppListActivity.java:144)
org.odk.collect.android.activities.FormDownloadList.onSaveInstanceState (FormDownloadList.java:345)
android.app.Activity.performSaveInstanceState (Activity.java:1388)
android.app.Instrumentation.callActivityOnSaveInstanceState (Instrumentation.java:1286)
android.app.ActivityThread.callCallActivityOnSaveInstanceState (ActivityThread.java:4616)
android.app.ActivityThread.handleRelaunchActivity (ActivityThread.java:4570)
android.app.ActivityThread.access$1000 (ActivityThread.java:181)
android.app.ActivityThread$H.handleMessage (ActivityThread.java:1488)
android.os.Handler.dispatchMessage (Handler.java:102)
android.os.Looper.loop (Looper.java:145)
android.app.ActivityThread.main (ActivityThread.java:6145)
java.lang.reflect.Method.invoke (Method.java)
```

**Comment by lognaturel** [comment]
@shobhitagarwal1612 any idea why `searchView` would be null in those cases? Should we include a `null` check or is there a better way to avoid it?

**Comment by lognaturel** [comment]
Devices affected are Samsung Galaxy A5, Mediatek MT6580, Qualcomm MSM8909. I suspect it has to do with hardware menu buttons because that's when `onCreateOptionsMenu` isn't called. However, neither @shobhitagarwal1612 nor @yanokwa could reproduce on emulators with hardware buttons.

Let's check for null to avoid a crash, log when it's null and see if we can get any more info that way.

**Comment by lognaturel** [comment]
@mmarciniak90 I believe you have a physical device with a hardware menu button! If you do, could you please try going to one of the views that has a list of forms and then rotating the orientation to see if you can get a crash? 🙏

**Comment by mmarciniak90** [comment]
I was able to reproduce this crash on only one device - Samsung Young with Android 4.1.
This is device with the worst specs.
To reproduce this crash I changed few times a device orientation. I done this quite fast, view on device could not turn around.
Crash is visible on views:
* Fill Bland Form
* Edit Saved Form
* View Sent Form
* Get Blank Form

Crash does not occur on views:
* Send Finalized Form
* Delete Saved Form

I tested this on ODK v1.10.0.
I checked also ODK v1.10.1 and there this crash was not visible.
